sdk/core/core-sse/src/utils.ts

```diff
@@ -72,7 +74,11 @@ export function ensureAsyncIterable(stream: NodeJS.ReadableStream | ReadableStre
     return {
       cancel: async () => {
         // drain the stream
-        stream.resume();
+        if (typeof (stream as Readable).destroy === "function"){
+          (stream as Readable).destroy();
+        } else {
+          stream.resume();
+        }
```

Comment:
This is needed because we don't want the underlying http socket to hold up the event loop. I initially mistakenly thought `resume` will be enough but it is not.